### ISSUE TITLE
Dark chunk when world is saved (caused by Radon)

### ISSUE THREAD

**Issue by Daverkex** [CyclopsMC/IntegratedDynamics#1254]
#### Issue type:

- :bug: Bug 

____

#### Short description:

When a chunk with a network exits the possibility of generating a dark chunk with no natural light

#### Steps to reproduce the problem:

1. Create a creative redstone ready map
2. Create a small network in only one chunk
3. Exist the map
4. Enter again
5. Repeat if not happen on the 1st try

#### Expected behaviour:

No dark chunk when the world loads again

#### Hotfix:
Fill the high of the chunk with a block and remove again solving the issue per block, not in all the chunk. It's necessary to fill the 16x16 and remove it. It isn't necessary to fill all the high only to override a "ghost block" in several point.

____

#### Versions:

- This mod: 1.16.2
- Minecraft: 1.19.2
- Forge: 43.2.6

___

## This is using [ATM8 modpack](https://www.curseforge.com/minecraft/modpacks/all-the-mods-8). The issue is reported in other places like:
* https://www.reddit.com/r/allthemods/comments/105nqyv/atm8_weird_darkness_appearing_in_this_chunk_on_my/
* https://www.reddit.com/r/allthemods/comments/z3y6sa/atm8_need_help_with_this_dark_spot/
* https://www.reddit.com/r/allthemods/comments/zyc9f8/having_this_weird_visual_bug_on_my_atm8_world_how/
* https://www.reddit.com/r/allthemods/comments/1226g8i/help_atm8_server_has_broken_chunks/

### **I discovered that happens putting networks of this mod in the chunk**

**I can't ensure that happen for a bug in this mod** but I want to relate the issue in the modpack repo to this one. I think it's a good point to start tracking the issue.

The modpack issue: https://github.com/AllTheMods/ATM-8/issues/1182

**Comment by rubensworks** [comment]
Thanks for reporting!

**Comment by rubensworks** [comment]
Can you try reproducing the problem with any other mods installed? Just so that we can rule out other mods?

**Comment by Daverkex** [comment]
**I couldn't reproduce** in an instance only with the Integrated Dynamics mods:

- Minecraft 1.19.2
- Forge 43.2.0
- IntegratedTunnels 1.8.18
- IntegratedDynamics 1.16.2
- Integrated NBT 1.6.0
- IntegrateCrafting 1.1.1
- IntegratedTerminals 1.4.5

**Comment by rubensworks** [comment]
Thanks for checking!

Then either it is a) another mod that causes the problem, or b) a combination of this mod and other mod(s) that cause the problem.
In any case, it might be better to follow this up first with your pack creator in https://github.com/AllTheMods/ATM-8/issues/1182.

Closing this one here in the meantime, but happy to re-open once new information would arise that indicates something needs to be changed in ID.

**Comment by Mitchell5200** [comment]
ATM pack dev here, just a heads up that i've managed to reproduce this in an isolated install when adding Radon into the mix, as that mod affects the lighting engine, i believe there's an incompatibility here affecting/causing this issue.
![2023-04-05_22 50 35](https://user-images.githubusercontent.com/45913837/230119907-f4cb714b-663b-45d4-add7-e9d905bc4e3b.png)

I also want to note that ID is not the only mod i've seen with a dark chunk issue within ATM8, Hexerei also had one, however i believe that was an issue traced to within hexerei and has been fixed in 0.3.0 of that mod

**Comment by rubensworks** [comment]
Thanks for the input @Mitchell5200!
Good to know that combining Radon with ID causes the problem.
So there must be something that Radon is expecting from ID, which is not happening.

Do you have any pointers to what Hexerei changed that fixed this issue?

**Comment by Mitchell5200** [comment]
at a guess, either this commit that references a chunks issue
https://github.com/JoeFoxe/Hexerei-1.19/commit/a4f55404560febdcf207baf07c4b0dfe76c16e56

or the 0.3.0 update commit with almost 1200 file changes
https://github.com/JoeFoxe/Hexerei-1.19/commit/f862b173f6a943d42112e5c472b4d0d4b897b0a8

sorry i cannot be of more help there

**Comment by rubensworks** [comment]
I see the first commit is by @EpicSquid.
@EpicSquid, do you perhaps know how this issue was handled in Hexerei?

**Comment by EpicSquid** [comment]
What's this about? I need more context currently.

**Comment by EpicSquid** [comment]
Ah sorry didn't see the earlier replies, mobile is a pain. I can take a better look in a few days more than likely, or contact Joe foxe in the hexerei discord.

**Comment by rubensworks** [comment]
@JoeFoxe Do you have any insights from your end on the above?

**Comment by rubensworks** [comment]
@Asek3 Since this issue only occurs when Radon is installed, I was wondering if you have any insights into why lighting is breaking with ID cables? Is this an issue that can be fixed on Radon's end?

**Comment by Twisted-Code** [comment]
Wanted to add my observations about this bug since it happened to me in ATM8 a few weeks ago. For me it occurred, initially, with a network that only had (I think) 2 cables and a battery which I was trying to use to power 3 machines (the first 2 directly on the third through Integrated Tunnels' energy interface and energy exporter). Actually the battery might've technically been a second network since it was only connected to the interface, but I guess that depends on whether I had placed the interface when the bug happened.
I don't know how useful this observation will be to you, since it looks like you may have already gotten much further than I did did in identifying the culprit.... but I'm trying to help, that counts for something right?

**Comment by rubensworks** [comment]
Since I'm getting no response from the Radon developers, there's not much I can do on my end unfortunately, other than recommending not to use Radon together with this mod.

Should anyone have any more information on how Radon changes vanilla lighting, how this could break with ID's cables, and how (if even) ID could work around this, please let me know, and I'll happily re-open this issue.

**Comment by TheAdamizer** [comment]
Hey there I was pretty frustrated with this bug so I did a little testing; maybe it's helpful enough to move forward on this.
I started fiddling with some of the ID config and found that I could set the `common > machine > cable > dynamicShape` setting to `off` and that *mostly* resolved the issue.  It seems like dynamic lighting is still kinda weird as the chunk seems 'dimmer' than it was before, but placing torches now properly inhibits spawns and the light level increases near the torch as expected.

**Comment by TheAdamizer** [comment]
Ok so I noticed this issue opened on the ATM8 repo where they mentioned they will be switching from Radon to Starlight for lighting engine optimizations, and under starlight this bug is essentially resolved HOWEVER anyone who ends up here trying to fix their (older) atm8 instance should read my comment there; I found a way to break loading of a chunk indefinitely that seems to be related to the Invisible light blocks ID's static light panels use.  I would say be careful if you have run into this bug already to avoid a crash.
To note, it looks like the invisible light block keeps hanging around unexpectedly when placing a panel and using ID + Starlight together.  I'd recommend not using them for your lighting needs if this applies to you.
